Jenkinsfile

```diff
@@ -17,42 +17,55 @@ spec:
     tty: true
 """
 
+String skipciMessage = 'Aborting because commit message contains [skip ci]'
+
 timestamps {
   ansiColor('gnome-terminal') {
     podTemplate(yaml: podDefinition) {
       node(POD_LABEL) {
-        stage('Checkout scm') {
-            checkout scm
-        }
-        stage('Import Scilla') {
-            container('scilla') {
-                sh "mkdir ./moving_folder"
-                sh "cp -r /scilla ./moving_folder/scilla"
-            }
-            container('ubuntu') {
-                sh "cp -r ./moving_folder/scilla /scilla"
-                sh "rm -rf ./moving_folder"
-                sh "ls -la /scilla/0"
-            }
-        }
-        container('ubuntu') {
-            env.VCPKG_ROOT="/vcpkg"
-            stage('Configure environment') {
-                sh "./scripts/setup_environment.sh"
-                // sh "git clone https://github.com/microsoft/vcpkg ${env.VCPKG_ROOT}"
-                // sh "export VCPKG_FORCE_SYSTEM_BINARIES=1 && cd ${env.VCPKG_ROOT} && git checkout 2022.07.25 && ${env.VCPKG_ROOT}/bootstrap-vcpkg.sh"
-            }
-            stage('Build') {
-                sh "git config --global --add safe.directory '*'"
-                sh "export VCPKG_ROOT=${env.VCPKG_ROOT} && ./scripts/ci_build.sh"
-            }
-            stage('Integration test') {
-                sh "scripts/integration_test.sh --setup-env"
-            }
-            stage('Report coverage') {
-                // Code coverage is currently only implemented for GCC builds, so OSX is currently excluded from reporting
-                sh "./scripts/ci_report_coverage.sh"
-            }
+        try {
+          stage('Checkout scm') {
+              checkout scm
+              def skipci = sh(script: "git log -1 --pretty=%B | fgrep -ie '[skip ci]' -e '[ci skip]' | wc -l", returnStdout: true).trim()
```

Comment:
So, how does this work for merge commits of PR with master that are being checked?  The problem is that even if I put [skip-ci] in the history, the merge commit will have a commit message "merge ... blah blah," and will likely not mention the commit message of the original commit.

We either need to check the child commit of the merge commit *that is not master*, but also exlcude this logic if this commit is *not* merge commit (if we fail to do the latter, [skip ci] might have an effect on follow-on PRs.

Therefore my suggestion was to look at the PR title.

Reply:
got it @valeryz, thanks for the feedback. Reviewing it

Reply:
I included in the condition the check of the PR title as per PR 3034. @valeryz @frankmeds